sabnzbd/config.py

```diff
@@ -435,10 +438,17 @@ def set_dict(self, values: Dict[str, Any]):
             "enable",
             "optional",
             "retention",
+            "expire_date",
+            "quota_left",
+            "usage_at_start",
             "priority",
             "notes",
         ):
             try:
+                if kw == "usage_at_start":
+                    usage_at_start_new = values.get("usage_at_start_new", "")
+                    if usage_at_start_new and values["quota_left"] != values["quota_left_old"]:
+                        values["usage_at_start"] = usage_at_start_new
```

Comment:
I suggest we move this out of the `for` loop and not rely on information stored inside the HTML but just do the logic here if we need to update. It can be an if-statement below the `if not self.displayname():` one.

Reply:
I first tried this:
```
servers = get_servers()
server = servers[self.__name]
logging.debug("quota_left_old: %s, from server: %s", values.get("quota_left_old", ""), server.quota_left())
```
When I do that, it crashes immediately:
```
$ python SABnzbd.py -f sabnzbd.ini -l 2 -b 0
Traceback (most recent call last):
  File "SABnzbd.py", line 1742, in <module>
    main()
  File "SABnzbd.py", line 995, in main
    res, msg = config.read_config(inifile)
  File "sabnzbd\sabnzbd\decorators.py", line 36, in call_func
    return f(*args, **kw)
  File "sabnzbd\sabnzbd\config.py", line 767, in read_config
    return _read_config(path)
  File "sabnzbd\sabnzbd\config.py", line 839, in _read_config
    ConfigServer(server.replace("{", "[").replace("}", "]"), CFG["servers"][server])
  File "sabnzbd\sabnzbd\config.py", line 421, in __init__
    self.set_dict(values)
  File "sabnzbd\sabnzbd\config.py", line 429, in set_dict
    logging.debug("quota_left_old: %s, from server: %s", values.get("quota_left_old", ""), server.quota_left())
AttributeError: 'dict' object has no attribute 'quota_left'
```
I tried `server.get("quota_left", "")` instead, but then I get `AttributeError: 'ConfigServer' object has no attribute 'get'` when I try to save. If you know how to solve this I can remove `quota_left_old` and `usage_at_start` from the form.

If you know how to get `sabnzbd.BPSMeter.grand_total` from the same place I can remove `usage_at_start_new` as well.

Reply:
Never mind, I figured it out. I assume it's too late to do it after the loop because then the old `quota_left` and `usage_at_start` have been replaced by the new values in the server object.